src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -541,7 +541,8 @@ public void completeTask(String taskId, AccessControlResponse accessControlRespo
 
         //Lock & update Task
         TaskResource task = findByIdAndObtainLock(taskId);
-        taskHasCompleted = task.getState() == CFTTaskState.COMPLETED;
+        taskHasCompleted = task.getState() != null
+            && (task.getState().equals(CFTTaskState.COMPLETED) || task.getState().equals(CFTTaskState.TERMINATED));
```

Comment:
I think there's a slight edge case here which we should cater for.  If a Task is TERMINATED then it could have been completed or cancelled previously and this info is in the termination_reason field.  So how do we deal with the case where a request is to complete a Task but it has been Cancelled?  I think the current behaviour doesn't cater for this also.
I think it should be an error although can discuss the details.